### ISSUE TITLE
synchronize specific files

### ISSUE THREAD

**Issue by ychaouche** [lsyncd/lsyncd#674]
### Sample config

```
sync {
  default.rsyncssh,     
  source	= "/home/ychaouche/.bashrc_common",
  host		= "10.10.10.19",
  targetdir	= "~/",
}
```

### Error
```
ychaouche#ychaouche-PC 14:11:07 ~ $ lsyncd /etc/lsyncd.conf
Error: cannot get absolute path of dir '/home/ychaouche/.bashrc_common': is not a directory
Error: Cannot access source directory: /home/ychaouche/.bashrc_common
ychaouche#ychaouche-PC 14:13:06 ~ $ 

```

### Question
Can I somehow make syncs of specific files?

**Comment by solracsf** [comment]
You can use `rsync` params `--include` or `--include-from`
https://linux.die.net/man/1/rsync

See example here how to pass `rsync` params
https://lsyncd.github.io/lsyncd/manual/config/layer4/

**Comment by ychaouche** [comment]
Thanks for the feedback @solracsf 
What I ended up doing is putting files I wanted to sync inside a SYNC directory, and have their original locations be symlinks to their new location inside the SYNC directory. So for example : 

```
$ mkdir SYNC
$ mv .bashrc_common SYNC/
$ ln -s ~/SYNC/.bashrc 
$ mv .emacs.d/init.el SYNC/
$ ln -s ~/SYNC/init.el .emacs.d/

```
I then synchronize the SYNC directory with remote servers.

```
sync {
  default.rsyncssh,     
  source	= "/home/ychaouche/SYNC",
  host		= "10.10.10.19",
  targetdir	= "/root/",
}

```

And on each remote server, I replace the original files with symlinks to their updated version inside SYNC. So

```
$ mv .bashrc_common .bashrc_common-
$ ln -s /root/SYNC/.bashrc_common
$ mv .emacs.d/init.el .emacs.d/init.el-
$ ln -s /root/SYNC/init.el .emacs.d/

```

It's a little bit cumbersome but semantically closer to what I'd like to do since a couple of the synchronization programs I checked (mainly _csync2_, _clsync_ and _lsyncd_) monitor whole directories, not single files. Maybe this is a limitation of the underlying _inotify_ mechanism.
